src/main/kotlin/mate/academy/service/FigureSupplier.kt

```diff
@@ -0,0 +1,33 @@
+package mate.academy.service
+
+import mate.academy.model.Circle
+import mate.academy.model.Figure
+import mate.academy.model.IsoscelesTrapezoid
+import mate.academy.model.Rectangle
+import mate.academy.model.RightTriangle
+import mate.academy.model.Square
+import mate.academy.util.randomFullRangeInt
+import mate.academy.util.randomHighRangeInt
+import mate.academy.util.randomLowRangeInt
+import kotlin.random.Random.Default.nextInt
+
+class FigureSupplier {
+
+    fun getRandomFigure(): Figure {
+        val randomIndex = nextInt(Figure::class.sealedSubclasses.size)
+        val randomColor = ColorSupplier().getRandomColor()
+
+        return when (Figure::class.sealedSubclasses[randomIndex]) {
```

Comment:
Interesting solution.

You also may use Enum for this goal. or Kotlin's random() functionality

Reply:
done, used random fn